cluster/log-dump/log-dump.sh

```diff
@@ -0,0 +1,344 @@
+#!/bin/bash
```

Comment:
Not deleting the old log-dump.sh just yet because then the current kubetest (which shells out to the script) will fail. After this PR is pushed, I'll reflect the change in test-infra and push a new image. Then we can delete it.